### ISSUE TITLE
PWA Discord Web

### ISSUE THREAD

**Issue by ThaUnknown** [Vencord/plugin-requests#50]
### Plugin Description

A plugin which makes discord.com/app installable as a PWA, this would require embedding a webmanifest in the head element, and display notifications by patching `setPageTitleNotificationCount` and instead using `navigator.setAppBadge`, maybe also modify the title/menu bar?

the manifest would look smth like:
```js
{
  name: 'Discord',
  short_name: 'Discord',
  start_url: 'https://discord.com/channels/@me', // URL when PWA launches
  display: 'fullscreen',
  display_override: ['window-controls-overlay'],
  lang: 'en-US',
  background_color: '#202225',
  theme_color: '#202225',
  scope: 'https://discord.com', // scope of all possible URL's
  description: 'Imagine a place...',
  orientation: 'landscape',
  icons: [{
      src: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQAAAAEAC..........',
      sizes: '256x256',
      type: 'image/png'
    }]
}
```

### Existing Plugin for other mod

existed for GM, doesn't work anymore: https://github.com/ThaUnknown/GMPlugins/blob/main/src/PWAify/index.js

### Request Agreement

- [X] I read the above and made sure my plugin is possible and allowed
- [X] I checked the [existing plugins](https://vencord.dev/plugins) and made sure my plugin doesn't already exist
- [X] I [searched the existing issues](https://github.com/Vencord/plugin-requests/issues?q=is%3Aissue) and made sure a similar request doesn't already exist

**Comment by Riddim-GLiTCH** [comment]
Isnt this space already fufilled by [Vesktop](https://github.com/Vencord/Vesktop)?

**Comment by ThaUnknown** [comment]
vesktop is fundamentally no different from discord, the idea is to not need to use electron but still have a somewhat native experience like shortcut, standalone window and launch on boot

**Comment by RokeJulianLockhart** [comment]
> https://github.com/Vencord/plugin-requests/issues/50#issuecomment-1947420521

Indeed, @ThaUnknown, I want to be able to utilize my extensions other than Vesktop. Additionally, with accessibility needs, shall need a PWA to be able to modify the font, for instance.

**Comment by Riddim-GLiTCH** [comment]
You can modify the font in desktop with CSS.

**Comment by nin0-dev** [comment]
also- you don't need PWAs to make a site installable as a window on most browsers

**Comment by ThaUnknown** [comment]
> nin0

yes, but this request isn't "make it installable as a PWA", but "make it usable as a PWA", there are other things like notification icon, screenshare fixes, menu bar etc

**Comment by Riddim-GLiTCH** [comment]
Just use vesktop at that point then.

**Comment by ThaUnknown** [comment]
the entire point of this, is to not use a standalone electron instance.....
